dataland-backend/src/main/kotlin/org/dataland/datalandbackend/service/DataManager.kt

```diff
@@ -0,0 +1,156 @@
+package org.dataland.datalandbackend.service
+
+import org.dataland.datalandbackend.edcClient.api.DefaultApi
+import org.dataland.datalandbackend.interfaces.DataManagerInterface
+import org.dataland.datalandbackend.model.CompanyMetaInformation
+import org.dataland.datalandbackend.model.DataManagerInputToGetData
+import org.dataland.datalandbackend.model.DataMetaInformation
+import org.dataland.datalandbackend.model.StorableDataSet
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+/**
+ * Implementation of a data manager for Dataland including meta data storages
+ */
+@Component("DefaultManager")
+class DataManager(
+    @Autowired var edcClient: DefaultApi
+) : DataManagerInterface {
+    var dataMetaData = mutableMapOf<String, DataMetaInformation>()
+    var companyData = mutableMapOf<String, CompanyMetaInformation>()
+    private var companyCounter = 0
+
+    /*
+    ________________________________
+    Helper-methods:
+    ________________________________
+     */
+
+    private fun verifyCompanyIdExists(companyId: String) {
+        if (!companyData.containsKey(companyId)) {
+            throw IllegalArgumentException("Dataland does not know the company ID $companyId.")
+        }
+    }
+
+    private fun verifyDataIdIsExists(dataId: String) {
+        if (!dataMetaData.containsKey(dataId)) {
+            throw IllegalArgumentException("Dataland does not know the data ID: $dataId.")
+        }
+    }
+
+    private fun verifyDataTypeExists(dataType: String) {
+        val matchesForDataType = dataMetaData.filter { it.value.dataType.equals(dataType) }
+        if (matchesForDataType.isEmpty()) {
+            throw IllegalArgumentException("Dataland does not know the data type: $dataType")
+        }
+    }
+
+    /*
+    ________________________________
+    Methods to route data inserts and queries to the data store and save meta data in the Dataland-Meta-Data-Storage:
+    ________________________________
+     */
+
+    override fun addDataSet(storableDataSet: StorableDataSet): String {
+        verifyCompanyIdExists(storableDataSet.companyId)
+
+        val dataId = edcClient.insertData(storableDataSet.data)
```

Comment:
hatten wir nicht besprochen dass in EuroDaT alles gespeichert werden soll, um aus den EuroDaT Informationen wieder unsere Daten aufbauen zu können? Sollte es dann nicht `edcClient.insertData(storableDataSet) sein, weil company und data type ja mit in EuroDaT gespeichert werden sollen?

Reply:
Korrekt.

Aber sollte ich dann nicht erst den EDC-Dummyserver dahingehend umschreiben, dass er auch companyId und dataType mitspeichert? Tut er aktuell noch nicht.

Und danach hier umstellen, dass alles mitgegeben wird?

Reply:
ja, bitte :-)

Reply:
Fixed

Reply:
ok